public/javascripts/study/tour-chapter.js

```diff
@@ -89,3 +89,8 @@ lichess.studyTourChapter = function (study) {
     tour.start();
   });
 };
+lichess.cancelCurrentTour = function () {
+  if (typeof Shepherd !== 'undefined' && Shepherd.activeTour) {
+    Shepherd.activeTour.cancel();
+  }
+};
```

Comment:
Need the initial `typeof` check since this function is being called when the user opens the "new chapter" dialog (to close the normal study tour if it exists). That can happen before Shepherd has been loaded so we do this check.